rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +486,337 @@ def accepted_type(self) -> type:
+                and isinstance(field.typeval, OpaqueValue)
+                and field.typeval.size != field.length.value
+            ):
+                field.first = UNDEFINED
+                field.length = UNDEFINED
+                field.typeval.clear()
+                break
+
+            self._last_field = nxt
+            nxt = self._next_field(nxt)
+
+    def get(self, fld: str) -> Any:
+        if fld not in self.valid_fields:
+            raise ValueError(f"field {fld} not valid")
+        return self._fields[fld].typeval.value
+
+    @property
+    def bitstring(self) -> Bitstring:
+        bits = ""
+        field = self._next_field(INITIAL.name)
+        while field and not field == FINAL.name:
+            field_val = self._fields[field]
+            if (
+                not field_val.set
+                or not isinstance(field_val.first, Number)
+                or not field_val.first.value <= len(bits)
+            ):
+                break
+            bits = bits[: field_val.first.value] + str(self._fields[field].typeval.bitstring)
+            field = self._next_field(field)
+        if len(bits) % 8:
+            raise ValueError(f"message length must be dividable by 8 ({len(bits)})")
+
+        return Bitstring(bits)
+
+    @property
+    def value(self) -> Any:
+        raise NotImplementedError
+
+    @property
+    def bytestring(self) -> bytes:
+        bits = str(self.bitstring)
+        return b"".join(
+            [int(bits[i : i + 8], 2).to_bytes(1, "big") for i in range(0, len(bits), 8)]
+        )
+
+    @property
+    def fields(self) -> List[str]:
+        return [f.name for f in self._model.fields]
+
+    @property
+    def accessible_fields(self) -> List[str]:
+
+        nxt = self._next_field(INITIAL.name)
+        fields: List[str] = []
+        while nxt and nxt != FINAL.name:
+
+            if (
+                self.__simplified(self._model.field_condition(Field(nxt))) != TRUE
+                or not self._has_first(nxt)
+                or (
+                    not self._has_length(nxt)
+                    if not isinstance(self._fields[nxt].typeval, OpaqueValue)
+                    else not self._is_valid_opaque_field(nxt)
+                )
+            ):
+                break
+
+            fields.append(nxt)
+            nxt = self._next_field(nxt)
+        return fields
+
+    def _is_valid_opaque_field(self, field: str) -> bool:
+
+        if self._get_length_unchecked(field) == UNDEFINED:
+            return False
+
+        for edge in self._model.incoming(Field(field)):
+            if self.__simplified(edge.condition) == TRUE:
+                valid_edge = edge
+                break
+        else:
+            return True
+
+        for ve in valid_edge.length.variables():
+            assert isinstance(ve.name, str)
+            if ve.name in self._fields and not self._fields[ve.name].set:
+                return False
+
+            if Last("Message"):
+                return True
```

Comment:
That is definitely wrong. Why is this not found by the unit tests? Is it even needed?

1. Either remove this condition or, if it is actually needed, add a unit test covering this case.
2. Please fix the currently failing unit test: test_array_type_spec (tests.test_parser.TestParser). Ideally you should ensure all unit tests run successfully before adding further commits to a pull request.

Reply:
Fixed by c41781e